lib/core/network/api/aws_amplify/amplify_service.dart

```diff
@@ -0,0 +1,282 @@
+      print('CognitoUserMfaRequiredException $e');
+      // handle SMS_MFA challenge
+    } on CognitoUserSelectMfaTypeException catch (e) {
+      print('CognitoUserSelectMfaTypeException $e');
+      // handle SELECT_MFA_TYPE challenge
+    } on CognitoUserMfaSetupException catch (e) {
+      print('CognitoUserMfaSetupException $e');
+      // handle MFA_SETUP challenge
+    } on CognitoUserTotpRequiredException catch (e) {
+      print('CognitoUserTotpRequiredException $e');
+      // handle SOFTWARE_TOKEN_MFA challenge
+    } on CognitoUserCustomChallengeException catch (e) {
+      print('CognitoUserCustomChallengeException $e');
+
+      // handle CUSTOM_CHALLENGE challenge
+      final loginCode = e.challengeParameters['loginCode'];
+      await eosService.loginWithCode(accountName: accountName, loginCode: loginCode, network: Networks.telos);
+      print('return challenge $loginCode');
+      session = await cognitoUser.sendCustomChallengeAnswer(loginCode);
+
+      return true;
+    } on CognitoUserConfirmationNecessaryException catch (e) {
+      print('CognitoUserConfirmationNecessaryException $e');
+      // handle User Confirmation Necessary
+    } on CognitoClientException catch (e) {
+      print('CognitoClientException $e');
+      // handle Wrong Username and Password and Cognito Client
+    } catch (e) {
+      print('error logging in: $e');
+    }
+
+    return true;
+  }
+
+  Future<ProfileData> getProfile() async {
+    final result = await _request(
+      path: 'get-profile',
+      body: {
+        'originAppId': remoteConfigService.pppOriginAppId,
+      },
+    );
+    final Map<String, dynamic> data = result['profile'];
+    final ProfileData profile = ProfileData.fromPPPDataJson(data);
+    return profile;
+  }
+
+  ///
+  /// register method is used to modify any user attributes such as name, bio, etc
+  ///
+  Future<dynamic> register(Map<String, dynamic> pppData) async {
+    final result = await _request(
+      path: 'register',
+      body: <String, dynamic>{
+        ...pppData,
+        'originAppId': remoteConfigService.pppOriginAppId,
+        'appData': {},
+      },
+    );
+    return result;
+  }
+
+  // maybe try to cache credentials?
+  Future<CognitoCredentials> getCredentials() async {
+    final credentials = CognitoCredentials(
+      remoteConfigService.identityPoolId,
+      userPool,
+    );
+    await credentials.getAwsCredentials(session?.getIdToken().getJwtToken());
+    return credentials;
+  }
+
+  Future<dynamic> _request({
+    required String path,
+    Map<String, dynamic>? body,
+    Map<String, String>? headers,
+    Map<String, String>? queryParams,
+    CognitoCredentials? credentials,
+  }) async {
+    final String region = awsLambdaRegion;
+    credentials ??= await getCredentials();
+    return awsAuthenticatedRequest(
+      credentials: credentials,
+      awsRegion: region,
+      endpoint: awsLambdaEndpoint,
+      path: path,
+      body: body,
```

Comment:
do we have a better return type for these or dynamic is it?

Reply:
Good question. Most of them are JSON maps. 

I suppose register returns a JSON object of the new profile, so we could update.

Reply:
https://github.com/hypha-dao/hypha_wallet/issues/87